target/linux/realtek/dts-5.10/rtl8380_engenius_ews2910p.dts

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "rtl838x.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "engenius,ews2910p", "realtek,rtl838x-soc";
+	model = "EnGenius EWS2910P";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x10000000>;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "factory-reset";
```

Comment:
The current pictures on the website just have a "reset" label. It would make more sense to me to use this as the (software) reset button, with `KEY_RESTART` as keycode.

Reply:
`KEY_RESTART` would cause the device to reboot on a short press. Or when trying to do a factory reset, and you drop the paperclip. That's extremely unintuitive as there isn't much of an indication that the device is rebooting or erasing the config. Or coming from the vendor firmware and expecting the button to only do a factory reset, one would find the device behavior completely changed.

So yes, I can make this `KEY_RESTART` if it moves the PR along, but I think it's not the best idea.

Reply:
Well, that's the standard behaviour in OpenWrt for a button labelled "reset".

If you also provide the LED aliases (see for example commit c4927747d25af59db8233a66a59fdf9e8c0e395d), `led-boot` will blink fast for 5s when booting. This is the moment the user can press any (GPIO) key to enter failsafe mode, which e.g. allows you to recover from a borked configuration change where you locked yourself out of the device. You could consider failsafe to be a "soft factory reset", since it uses the default OpenWrt configuration (until reboot).

With `BTN_0`, button presses wouldn't have any effect in OpenWrt, unless the user made a custom script to handle these events.